plugin/storage/cassandra/dependencystore/storage.go

```diff
@@ -26,60 +26,110 @@ import (
 	casMetrics "github.com/jaegertracing/jaeger/pkg/cassandra/metrics"
 )
 
+// IndexMode determines how the dependency data is indexed.
+type IndexMode int
+
+// IsValid returns true if the IndexMode is a valid one.
+func (i IndexMode) IsValid() bool {
+	return i < end
+}
+
 const (
-	depsInsertStmt = "INSERT INTO dependencies(ts, ts_index, dependencies) VALUES (?, ?, ?)"
-	depsSelectStmt = "SELECT ts, dependencies FROM dependencies WHERE ts_index >= ? AND ts_index < ?"
+	// SASIEnabled is used when the dependency table is SASI indexed.
+	SASIEnabled IndexMode = iota
+
+	// SASIDisabled is used when the dependency table is NOT SASI indexed.
+	SASIDisabled
+	end
+
+	depsInsertStmtSASI = "INSERT INTO dependencies(ts, ts_index, dependencies) VALUES (?, ?, ?)"
+	depsInsertStmt     = "INSERT INTO dependencies_v2(ts, ts_bucket, dependencies) VALUES (?, ?, ?)"
+	depsSelectStmtSASI = "SELECT ts, dependencies FROM dependencies WHERE ts_index >= ? AND ts_index < ?"
+	depsSelectStmt     = "SELECT ts, dependencies FROM dependencies_v2 WHERE ts_bucket IN ? AND ts >= ? AND ts < ?"
+
+	// TODO: Make this customizable.
+	tsBucket = 24 * time.Hour
+)
+
+var (
+	errInvalidIndexMode = errors.New("invalid index mode")
 )
 
 // DependencyStore handles all queries and insertions to Cassandra dependencies
 type DependencyStore struct {
 	session                  cassandra.Session
 	dependenciesTableMetrics *casMetrics.Table
 	logger                   *zap.Logger
+	indexMode                IndexMode
 }
 
 // NewDependencyStore returns a DependencyStore
 func NewDependencyStore(
 	session cassandra.Session,
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
-) *DependencyStore {
+	indexMode IndexMode,
+) (*DependencyStore, error) {
+	if !indexMode.IsValid() {
+		return nil, errInvalidIndexMode
+	}
 	return &DependencyStore{
 		session:                  session,
 		dependenciesTableMetrics: casMetrics.NewTable(metricsFactory, "dependencies"),
 		logger:                   logger,
-	}
+		indexMode:                indexMode,
+	}, nil
 }
 
 // WriteDependencies implements dependencystore.Writer#WriteDependencies.
 func (s *DependencyStore) WriteDependencies(ts time.Time, dependencies []model.DependencyLink) error {
 	deps := make([]Dependency, len(dependencies))
 	for i, d := range dependencies {
-		deps[i] = Dependency{
+		dep := Dependency{
 			Parent:    d.Parent,
 			Child:     d.Child,
 			CallCount: int64(d.CallCount),
 		}
+		if s.indexMode == SASIDisabled {
```

Comment:
another argument to refer to this change as "dependencies v2" rather than tying to SASI.